addons/mail/models/ir_actions_server.py

```diff
@@ -192,15 +203,50 @@ def _compute_activity_info(self):
             to_reset.activity_date_deadline_range_type = False
             to_reset.activity_user_type = False
         for action in (self - to_reset):
-            if action.activity_type_id.res_model and action.model_id.model != action.activity_type_id.res_model:
-                action.activity_type_id = False
-            if not action.activity_summary:
-                action.activity_summary = action.activity_type_id.summary
             if not action.activity_date_deadline_range_type:
                 action.activity_date_deadline_range_type = 'days'
             if not action.activity_user_type:
                 action.activity_user_type = 'specific'
 
+            if action.activity_plan_id:
+                action.activity_type_id = False
+                if not action.activity_plan_has_user_on_demand:
+                    action.activity_user_type = False
+                continue
+
+            if not action.activity_type_id or (
+                action.activity_type_id.res_model and action.model_id.model != action.activity_type_id.res_model
+            ):
+                action.activity_type_id = action.env['mail.activity']._default_activity_type_for_model(action.model_id.model)
+            action.activity_summary = action.activity_type_id.summary
+
+    @api.depends('model_id', 'state')
+    def _compute_activity_plan_available_ids(self):
+        self.activity_plan_available_ids = False
+        for action in self.filtered(lambda act: act.state == 'next_activity'):
+            action.activity_plan_available_ids = self.env['mail.activity.plan'].search(
+                fields.Domain.AND([
+                    ['|', ('res_model', '=', False), ('res_model', '=', action.model_id.model)],
+                    [('template_ids', '!=', False)],  # exclude plan without activities
+                ])
+            )
+
+    @api.depends('model_id', 'state', 'activity_type_id', 'activity_plan_available_ids')
```

Comment:
Compute depending on unstored field also here, bad idea. Also don't think you need activity_plan_available_ids dependency, as it is alread implied by other triggers (model notably). And if I choose a plan without templates by forcing the value and bypassing the domain limitatoin, well that's my right, it should not crash, just create nothing.